SongController: add SongController, DTOs y service interface

Palencia09 opened the pull request:
Porfavor revisar  😊👌

Palencia09 committed:
SongController: add SongController, DTOs y service interface

0xCamiX commented on src/main/java/com/slash/music/service/SongService.java:
Añadir java docs para los comentarios de cada una de las clases definidas en esta interfaz

0xCamiX commented on src/main/java/com/slash/music/dto/SongUpdateRequest.java:
Por qué el decorador @positive pero en el dto de songResponse no lleva?

0xCamiX commented on src/main/java/com/slash/music/controller/SongController.java:
Igualmente implementar java doc para los comentarios en las clases y añadir quién realizo la implementación, ver el módulo de artista para guía.

0xCamiX replied:
Utilizar inglés en los comentarios y en el proyecto por favor.

0xCamiX reviewed (changes requested):
Ajustar Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0. realiza un pull --rebase para obtener los últimos cambios del proyecto y revisar los estilos con Gradle antes de enviar la pr y comprobar la compilación correcta con ./gradlew test

DanielPenalozaB commented on src/main/java/com/slash/music/service/SongService.java:
```diff
@@ -0,0 +1,12 @@
+  Page<SongResponse> search(String title, String artist, int page, int size, String sort);
+  SongResponse update(Long id, SongUpdateRequest request);
+  void delete(Long id);
+}
```
El error ocurre porque `SongService` está definido como **interfaz,** pero no existe una **clase** concreta que lo **implemente.** Al iniciar los **tests,** `Spring` intenta crear el `SongController` y falla porque no encuentra ninguna clase de `SongService` que pueda instanciar.

La recomendación es definir `SongService` directamente como **clase,** tal como se hace con los demás **servicios** del proyecto, en lugar de mantenerlo como **interfaz** sin implementación. De esta manera, `Spring` podrá instanciar el servicio correctamente y el controlador funcionará sin errores en la ejecución de los tests.

Palencia09 committed:
fix(songs): corregir build/tests y alinear servicio/DTOs con entidad Song

- Implementa SongServiceImpl y registra @Service
- Ajusta a durationSeconds/fileUrl/genre según model.Song
- Crea ArtistRepository y AlbumRepository para resolver relaciones (artist obligatorio, album opcional)
- Extiende SongRepository con búsquedas por título y por artist.name
- Actualiza DTOs (Create/Update/Response) con durationSeconds, fileUrl, genre, artistId, albumId
- Corrige uso de ResourceNotFoundException según firma del proyecto
- Añade JavaDocs y comentarios en inglés (según review)
- Mantiene endpoints en /api/v1/songs

Refs: feedback de revisión y fallos de CI (build-and-test)

Palencia09 committed:
Se corrigen fallos de CI:
- Checkstyle: renombrado `findByArtist_NameContainingIgnoreCase` → `findByArtistNameContainingIgnoreCase` (camelCase).
- Indentación y llaves agregadas en `SongServiceImpl`/controller.
- Ajuste de llamada en `search(...)`.

Vuelvo a lanzar el pipeline con estos cambios.

Palencia09 commented:
@0xCamiX @DanielPenalozaB porfavor revisar, no se como resolver el error que salta al momento de Build Checks, cree ArtistRepository y AlbumRepository por conexiones que necesitaba, son 2 commits por revisar. Gracias 👌

Palencia09 committed:
Cambios en el formato para que apruebe el check porfavor probar @0xCamiX @DanielPenalozaB

DanielPenalozaB reviewed (approved):
LGTM

Palencia09 committed:
Cambios de public en los DTOs a Private

Palencia09 committed:
Add: AllArgsConstructor

Palencia09 committed:
Add: Contructor (Gets and Sets)
